### PR TITLE
Update rapids JNI and private dependency to 24.12.0-SNAPSHOT [skip ci]

### PR DESCRIPTION
**Pull request by nvauto**
To fix: https://github.com/NVIDIA/spark-rapids/issues/11492
Wait for the pre-merge CI job to SUCCEED

### CODE REVIEW

**Comment by nvauto** [comment]
build

**Comment by NvTimLiu** [comment]
build

File: jenkins/databricks/init_cudf_udf.sh [hunk 1]
```diff
@@ -20,8 +20,7 @@
 
 set -ex
 
-# TODO: https://github.com/NVIDIA/spark-rapids/issues/11492
-CUDF_VER=${CUDF_VER:-24.10}
+CUDF_VER=${CUDF_VER:-24.12
```
Comment (pxLi) [inline jenkins/databricks/init_cudf_udf.sh:23]: right `}`

**Review by pxLi — APPROVED** [review]

**Comment by pxLi** [comment]
build

**Pull request merged**